Sources/CreateOptions/PackageDeclaration.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+
+public enum SourceControlRequirement: Decodable {
+    case exact(version: String)
+    case range(from: String, to: String)
+    case closedRange(from: String, to: String)
+    case branch(name: String)
+    case commit(hash: String)
+    case from(version: String)
```

Comment:
I guess we are getting `Decodable` for free here by labelling the associated values? 

It would be nice if we could avoid having to nest the value for cases where there is only one associated value. For example, if this:

```yaml
- url: https://github.com/apple/swift-argument-parser
product: ArgumentParser
rule:
  exact:
    version: 1.1.1
- url: https://github.com/apple/swift-log
product: Logging
rule:
  branch:
    name: main
- url: https://github.com/apple/swift-numerics
product: Numerics
rule:
  commit:
    hash: 7f2d022d3d9b55bf812814f5d01896cbfa0fd4da
- url: https://github.com/apple/swift-system
product: SystemPackage
rule:
  from:
    version: 1.2.1
```

Could become this:

```yaml
- url: https://github.com/apple/swift-argument-parser
product: ArgumentParser
rule:
  exact: 1.1.1
- url: https://github.com/apple/swift-log
product: Logging
rule:
  branch: main
- url: https://github.com/apple/swift-numerics
product: Numerics
rule:
  commit: 7f2d022d3d9b55bf812814f5d01896cbfa0fd4da
- url: https://github.com/apple/swift-system
product: SystemPackage
rule:
  from: 1.2.1
```

No problem if not though

Reply:
We do get decodable for free however when an enum is decoded with an associated value (without name) it has to be of the format:
```yaml
rule:
  from:
	_0: 1.1.1
```

Please see the bottom of the PR summary for why this had to happen.

Reply:
> I started out aiming for this syntax however from a development (and therefore maintenance) perspective it was too complicated and heavy due to the decoding and using objects instead of an enum. What is implemented now is the simplest and another implementation would save only a single line per import, a negligible optimization.

Sorry yeah I think I read that but then forgot about it 😄 Lets keep the default Decodable implementation! Thanks